lib/Driver/Compilation.cpp

```diff
@@ -458,6 +458,26 @@ namespace driver {
       }
     }
 
+    /// Check to see if a batch-constituent job produced a zero-length
+    /// serialized diagnostics file, which is used to indicate constituents that
+    /// were batched together with a failing constituent but did not,
+    /// themselves, produce any errors.
+    bool batchConstituentJobProducedZeroLengthSerializedDiagnostics(
+      const Job *J) const {
+      auto DiaPath =
+        J->getOutput().getAnyOutputForType(file_types::TY_SerializedDiagnostics);
+      if (!DiaPath.empty()) {
```

Comment:
This formulation requires the reader to read many lines before understanding what happens if the DiaPath is empty. I'd be inclined to reverse the sense of the if, and use an early return.

Reply:
fixed

Reply:
Great!